<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ClusterAnalysisDashboard</class>
 <widget class="QDialog" name="ClusterAnalysisDashboard">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cluster Analysis Dashboard</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
/* Professional Dark Theme */
QDialog {
    background-color: #1E1E1E;
    color: #E1E1E1;
    font-family: 'Segoe UI', Arial, sans-serif;
    border: 1px solid #3A3A3A;
    border-radius: 8px;
}

QTableWidget {
    background-color: #252526;
    color: #CCCCCC;
    gridline-color: #3E3E42;
    border: 1px solid #464647;
    border-radius: 6px;
    font-size: 12px;
    selection-background-color: #094771;
    selection-color: #FFFFFF;
    alternate-background-color: #2A2A2B;
}

QTableWidget::item {
    padding: 12px 10px;
    border: none;
    border-bottom: 1px solid #3E3E42;
}

QTableWidget::item:hover {
    background-color: #2F2F30;
}

QTableWidget::item:selected {
    background-color: #094771;
    color: #FFFFFF;
}

QHeaderView::section {
    background-color: #37373D;
    color: #FFFFFF;
    padding: 14px 10px;
    border: none;
    font-weight: 600;
    font-size: 13px;
    border-right: 1px solid #3E3E42;
    border-bottom: 2px solid #007ACC;
}

QHeaderView::section:hover {
    background-color: #404040;
}

QHeaderView::section:first {
    border-top-left-radius: 6px;
}

QHeaderView::section:last {
    border-top-right-radius: 6px;
    border-right: none;
}

QPushButton {
    background-color: #0E639C;
    color: #FFFFFF;
    border: 1px solid #1177BB;
    padding: 12px 28px;
    font-size: 13px;
    font-weight: 500;
    border-radius: 5px;
    min-width: 100px;
}

QPushButton:hover {
    background-color: #1177BB;
    border: 1px solid #1E88E5;
}

QPushButton:pressed {
    background-color: #005A9E;
    border: 1px solid #0F4C75;
}

QPushButton:focus {
    outline: 2px solid #007ACC;
    outline-offset: 1px;
}

QPushButton#refreshButton {
    background-color: #0078D4;
}

QPushButton#refreshButton:hover {
    background-color: #106EBE;
}

QPushButton#exportButton {
    background-color: #107C10;
}

QPushButton#exportButton:hover {
    background-color: #0E6E0E;
}

QPushButton#identifySimilarButton {
    background-color: #7A2C7A;
}

QPushButton#identifySimilarButton:hover {
    background-color: #8B3A8B;
}

QPushButton#closeButton {
    background-color: #6B6B6B;
}

QPushButton#closeButton:hover {
    background-color: #5A5A5A;
}

QLabel#titleLabel {
    color: #FFFFFF;
    font-size: 18px;
    font-weight: 600;
    background-color: #2D2D30;
    padding: 20px;
    border-radius: 8px;
    border: 1px solid #3F3F46;
    margin-bottom: 15px;
}

QLabel#statusLabel {
    color: #CCCCCC;
    font-size: 12px;
    background-color: #2D2D30;
    padding: 8px 15px;
    border-radius: 4px;
    border: 1px solid #3F3F46;
}

QFrame#separatorFrame {
    background-color: #3E3E42;
    border: none;
}

QGroupBox {
    color: #FFFFFF;
    font-size: 14px;
    font-weight: 500;
    border: 2px solid #007ACC;
    border-radius: 8px;
    margin-top: 10px;
    padding-top: 10px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 8px 0 8px;
    background-color: #1E1E1E;
}
   </string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <property name="leftMargin">
    <number>20</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>20</number>
   </property>
   <property name="bottomMargin">
    <number>20</number>
   </property>
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="text">
      <string>Cluster Analysis Dashboard</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="controlsGroupBox">
     <property name="title">
      <string>Analysis Controls</string>
     </property>
     <layout class="QHBoxLayout" name="controlsLayout">
      <property name="spacing">
       <number>15</number>
      </property>
      <item>
       <widget class="QPushButton" name="refreshButton">
        <property name="text">
         <string>Refresh Analysis</string>
        </property>
        <property name="toolTip">
         <string>Refresh the cluster analysis data</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="identifySimilarButton">
        <property name="text">
         <string>Find Similar Clusters</string>
        </property>
        <property name="toolTip">
         <string>Identify and group similar clusters based on dimensions and volume</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="exportButton">
        <property name="text">
         <string>Export Data</string>
        </property>
        <property name="toolTip">
         <string>Export cluster analysis results to CSV file</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="closeButton">
        <property name="text">
         <string>Close</string>
        </property>
        <property name="toolTip">
         <string>Close the cluster analysis dashboard</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="dataGroupBox">
     <property name="title">
      <string>Cluster Data</string>
     </property>
     <layout class="QVBoxLayout" name="dataLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <item>
       <widget class="QTableWidget" name="clusterTable">
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
        <property name="cornerButtonEnabled">
         <bool>false</bool>
        </property>
        <attribute name="horizontalHeaderVisible">
         <bool>true</bool>
        </attribute>
        <attribute name="horizontalHeaderCascadingSectionResizes">
         <bool>false</bool>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>120</number>
        </attribute>
        <attribute name="horizontalHeaderHighlightSections">
         <bool>true</bool>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string>Cluster ID</string>
         </property>
         <property name="toolTip">
          <string>Unique identifier for the cluster or group</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Point Count</string>
         </property>
         <property name="toolTip">
          <string>Number of data points in this cluster</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Dimensions (X × Y × Z)</string>
         </property>
         <property name="toolTip">
          <string>Bounding box dimensions of the cluster</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Classification</string>
         </property>
         <property name="toolTip">
          <string>Type or grouping classification of the cluster</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Volume</string>
         </property>
         <property name="toolTip">
          <string>Calculated volume of the cluster bounding box</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="separatorFrame">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>2</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::HLine</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="statusLabel">
     <property name="text">
      <string>Ready to display cluster analysis results</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>closeButton</sender>
   <signal>clicked()</signal>
   <receiver>ClusterAnalysisDashboard</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>843</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>